core/tests/test-dependencies/root-files-mapping-generator.ts

```diff
@@ -0,0 +1,723 @@
+import {
+  project,
+  ROOT_DIRECTORY,
+  AngularDecorators,
+  getRelativePathToRootDirectory,
+  isNodeModule,
+  getDecorationNodesByTextFromSourceFile,
+  resolveModuleRelativeToRoot,
+  getDecorationNodeText,
+  getValueFromLiteralStringOrBinaryExpression,
+} from './typescript-ast-utilities';
+import {
+  getPageModules,
+  getPageModulesCoveredByRouteMapping,
+} from './route-to-module-mapping-generator';
+
+interface BaseAngularInformation {
+  className: string;
+}
+
+interface AngularModuleInformation extends BaseAngularInformation {
+  type: 'module';
+}
+
+interface AngularComponentInformation extends BaseAngularInformation {
+  type: 'component';
+  selector?: string;
+  templateFilePath?: string;
+}
+
+interface AngularDirectiveOrPipeInformation extends BaseAngularInformation {
+  type: 'directive' | 'pipe';
+  selector?: string;
+}
+
+type AngularInformation =
+  | AngularModuleInformation
+  | AngularComponentInformation
+  | AngularDirectiveOrPipeInformation;
+
+const GIT_IGNORED_EXCLUSIONS = fs
+  .readFileSync(path.resolve(ROOT_DIRECTORY, '.gitignore'), 'utf-8')
+  .split('\n')
+  .filter(line => line.trim() && !line.startsWith('#'));
+
+const FILE_EXCLUSIONS = [
+  ...GIT_IGNORED_EXCLUSIONS,
+  'types',
+  'typings',
+  'scripts',
+  'assets/scripts',
+  'core/tests/build_sources',
+  'core/tests/data',
+  'core/tests/load_tests',
+  'core/tests/release_sources',
+  'core/tests/services_sources',
+  'core/tests/test-dependencies',
+  'core/templates/tests',
+  'core/templates/services/UpgradedServices.ts',
+  'core/templates/services/angular-services.index.ts',
+  'core/templates/utility/hashes.ts',
+  'webpack.common.config.ts',
+  'webpack.common.macros.ts',
+  'webpack.dev.config.ts',
+  'webpack.dev.sourcemap.config.ts',
+  'webpack.prod.config.ts',
+  'webpack.prod.sourcemap.config.ts',
+  'angular-template-style-url-replacer.webpack-loader.js',
+];
+
+const FILE_EXTENSIONS = [
+  '.ts',
+  '.js',
+  '.html',
+  '.md',
+  '.css',
+  'CODEOWNERS',
+  'AUTHORS',
+  'CONTRIBUTORS',
+];
+
+const MANUALLY_MAPPED_DEPENDENCIES: Record<string, string[]> = {
+  '.lighthouserc-base.js': [
+    'puppeteer-login-script.js',
+    'core/tests/puppeteer/lighthouse_setup.js',
+  ],
+  'core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts': [
+    'core/tests/puppeteer-acceptance-tests/specs/helpers/reporter.ts',
+  ],
+  'core/templates/pages/header_css_libs.html': [
+    'core/templates/css/oppia.css',
+    'core/templates/css/oppia-material.css',
+  ],
+  'core/templates/pages/oppia-root/index.ts': [
+    'core/templates/pages/oppia-root/oppia-root.mainpage.html',
+  ],
+  'core/templates/pages/lightweight-oppia-root/index.ts': [
+    'core/templates/pages/lightweight-oppia-root/lightweight-oppia-root.mainpage.html',
+  ],
+  'core/templates/pages/error-pages/error-iframed-page/error-iframed-page.import.ts':
+    [
+      'core/templates/pages/error-pages/error-iframed-page/error-iframed.mainpage.html',
+    ],
```

Comment:
Why are these done manually?

Reply:
Most mainpage.html are mapped to their relative import file, but these have different filenames so they are manually mapped (they don't end with .import.ts) and the error-iframed-page.import.ts has a error-iframed.mainpage.html instead of error-iframed-page.mainpage.html.

Reply:
Hmm, for the error-iframed can we just fix the naming?

Reply:
Fixed the naming, Done.